### PR TITLE
Upgrade to VS 2017

### PR DESCRIPTION
**Pull request by natemcmaster**
Removes project.json and upgrades to MSBuild.

### CODE REVIEW

File: test/Microsoft.AspNetCore.Cors.Test/Microsoft.AspNetCore.Cors.Test.csproj [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+  <Import Project="..\..\build\common.props" />
+
+  <PropertyGroup>
+    <TargetFrameworks>netcoreapp1.1;net451</TargetFrameworks>
+  </PropertyGroup>
+
+  <ItemGroup>
+    <ProjectReference Include="..\WebSites\CorsMiddlewareWebSite\CorsMiddlewareWebSite.csproj" />
+    <ProjectReference Include="..\..\src\Microsoft.AspNetCore.Cors\Microsoft.AspNetCore.Cors.csproj" />
+    <PackageReference Include="Microsoft.NET.Test.Sdk" Version="15.0.0-*" />
```
Comment (pranavkm) [inline test/Microsoft.AspNetCore.Cors.Test/Microsoft.AspNetCore.Cors.Test.csproj]: Sort alphabetically.

File: test/Microsoft.AspNetCore.Cors.Test/Microsoft.AspNetCore.Cors.Test.csproj [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+    <PackageReference Include="xunit.runner.visualstudio" Version="2.2.0-*" />
+    <PackageReference Include="Microsoft.AspNetCore.TestHost" Version="1.2.0-*" />
+    <PackageReference Include="Microsoft.Extensions.Logging.Testing" Version="1.2.0-*" />
+    <PackageReference Include="Moq" Version="4.6.36-*" />
+    <PackageReference Include="xunit" Version="2.2.0-*" />
+  </ItemGroup>
+
+</Project>
```
Comment (pranavkm) [inline test/Microsoft.AspNetCore.Cors.Test/Microsoft.AspNetCore.Cors.Test.csproj]: VS kept adding the `<Service Include="{82a7f48d-3b50-4b1e-b82e-3ada8210c358}" />` item when I tried running tests . Could you verify if that's needed here and add it if necessary?

Comment (natemcmaster) [inline test/Microsoft.AspNetCore.Cors.Test/Microsoft.AspNetCore.Cors.Test.csproj]: I'm not sure why it is added. It's not required for dotnet-test. I've opened https://github.com/Microsoft/vstest/issues/409 to get clarification on what this does.

Comment (pranavkm) [inline test/Microsoft.AspNetCore.Cors.Test/Microsoft.AspNetCore.Cors.Test.csproj]: According to this SO response - http://stackoverflow.com/a/22917902, it's a VS thing.

**Pull request merged**